File: android/feature/ui/src/main/java/com/anyaicliremote/feature/ui/components/SessionStatusBar.kt
package com.anyaicliremote.feature.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

/**
 * Thin transient bar: an interaction-mode badge (plan/normal/yolo) and a passing
 * status notice (retry / auto model switch). Both are display-only, driven by
 * neutral session state; empty values render nothing.
 */
@Composable
fun SessionStatusBar(mode: String, notice: String, modifier: Modifier = Modifier) {
    val badge = modeBadge(mode)
    if (badge.isEmpty() && notice.isEmpty()) return
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 12.dp, vertical = 4.dp),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (badge.isNotEmpty()) {
            Text(
                text = badge,
                style = MaterialTheme.typography.labelSmall,
                fontWeight = FontWeight.SemiBold,
                color = MaterialTheme.colorScheme.onPrimaryContainer,
                modifier = Modifier
                    .clip(RoundedCornerShape(6.dp))
                    .background(MaterialTheme.colorScheme.primaryContainer)
                    .padding(horizontal = 8.dp, vertical = 2.dp),
            )
        }
        if (notice.isNotEmpty()) {
            Text(
                text = notice,
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

// Only surface a badge for modes worth flagging; plain "normal" needs no chrome.
private fun modeBadge(mode: String): String = when (mode.lowercase()) {
    "plan" -> "计划模式"
    "yolo" -> "YOLO 模式"
    "" , "normal", "default", "agent" -> ""
    else -> mode
}

File: ios/AnyAICLIRemoteFeature/Views/SessionStatusBar.swift
import SwiftUI

/// Thin transient bar: an interaction-mode badge (plan/yolo) and a passing status
/// notice (retry / auto model switch). Display-only; empty values render nothing.
struct SessionStatusBar: View {
  let mode: String
  let notice: String

  var body: some View {
    let badge = modeBadge(mode)
    if badge.isEmpty && notice.isEmpty {
      EmptyView()
    } else {
      HStack(spacing: 8) {
        if !badge.isEmpty {
          Text(badge)
            .font(.caption2.weight(.semibold))
            .padding(.horizontal, 8)
            .padding(.vertical, 2)
            .background(Color.accentColor.opacity(0.16), in: RoundedRectangle(cornerRadius: 6))
            .accessibilityIdentifier("session-mode-badge")
        }
        if !notice.isEmpty {
          Text(notice)
            .font(.caption2)
            .foregroundStyle(.secondary)
            .accessibilityIdentifier("session-status-notice")
        }
        Spacer(minLength: 0)
      }
      .padding(.horizontal, 12)
      .padding(.vertical, 4)
    }
  }

  // Only surface a badge for modes worth flagging; plain "normal" needs no chrome.
  private func modeBadge(_ mode: String) -> String {
    switch mode.lowercased() {
    case "plan": return "计划模式"
    case "yolo": return "YOLO 模式"
    default: return ""
    }
  }
}
